plugins/node/opentelemetry-koa-instrumentation/src/koa.ts

```diff
@@ -118,7 +121,8 @@ export class KoaInstrumentation extends BasePlugin<typeof koa> {
     middlewareLayer[kLayerPatched] = true;
     this._logger.debug('patching Koa middleware layer');
     return async (context: KoaContext, next: koa.Next) => {
-      if (api.getSpan(api.context.active()) === undefined) {
+      const parent = api.getSpan(api.context.active()) as KoaPluginSpan;
```

Comment:
i'm not sure why you are extending the span api, can't you use [`context()`](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-api/src/trace/span.ts#L42) instead ?

Reply:
I have to find out somehow who is the root parent span.. I do it by checking that the parentSpanId is undefined..

Reply:
I remembered that we do the same in the express plugin so thats fine

Reply:
This isn't safe. An instrumentation may be used by a third party SDK which doesn't have the same private properties we do. The safer (but more roundabout) way to check if it is the root would be to check if there is a span in context before the first span is created

Reply:
thanks @dyladan 
fixed